File: Flutter/Vistian.Remoting.Package/lib/src/invokeMessage.dart
import 'package:json_annotation/json_annotation.dart';
import 'message.dart';
import 'parameter.dart';
import 'serviceKey.dart';


part 'invokeMessage.g.dart';

@JsonSerializable(createFactory: false)
class InvokeMessage extends Message{
  @JsonKey(name:"Service")
  ServiceKey service;
  @JsonKey(name:"Method")
  String method;
  @JsonKey(name:"Parameters")
  List<Parameter> parameters;

  InvokeMessage(this.service,this.method,this.parameters);

  Map<String,dynamic> toJson() => _$InvokeMessageToJson(this) ;
}

File: Xamarin/Vistian.Flutter.Remoting.Core/InvokeMessage.cs
using System;
using System.Collections.Generic;

namespace Vistian.Flutter.Remoting
{
    /// <summary>
    /// Represents the invocation of a method for a specified instance .
    /// </summary>
    public class InvokeMessage
    {
        /// <summary>
        /// Gets or sets the key for the associated service.
        /// </summary>
        /// <value>The instance.</value>
        public ServiceKey Service { get; set; }

        /// <summary>
        /// Gets or sets the method name
        /// </summary>
        /// <value>The method.</value>
        public string Method { get; set; }

        /// <summary>
        /// Gets or sets the parameters.
        /// </summary>
        /// <value>The arguments.</value>
        public List<Parameter> Parameters { get; set; }
    }
}
